### PR TITLE
init project

### PR DESCRIPTION
**Pull request by horis233**
## init project
- copy the makefiles from meta-operator
- using operator sdk to generate api and controller template.
- leverage the MongoDB sts and service. (removed `priorityClassName: system-cluster-critical` and `runAsUser: 999` because the operator will not be in the kube- * namespace.)
Hard code the keyfile secret.

What we need to do
1. Need to generate `cluster-ca-secret`.
2. push images to quay.io currently, I use the artifactory with image-pull-secret 
```
kubectl -n ibm-mongodb-operator create secret docker-registry image-pull-secret --docker-server=hyc-cloud-private-edge-docker-local.artifactory.swg-devops.com/ibmcom-amd64 --docker-username=$DOCKER_USERNAME --docker-password=$DOCKER_PASSWORD
```

@chenzhiwei

### CODE REVIEW

File: common/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+image: __IMAGE_REPO__/__IMAGE_NAME__:__RELEASE_TAG__
```
Comment (chenzhiwei) [inline common/manifest.yaml]: we don't need this file.

File: deploy/crds/operator.ibm.com_v1alpha1_mongodb_cr.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: operator.ibm.com/v1alpha1
+kind: MongoDB
+metadata:
+  name: example-mongodb
+spec:
+  replicas: 3
+  pullSecret: image-pull-secret
+  imageRegistry: hyc-cloud-private-edge-docker-local.artifactory.swg-devops.com/ibmcom-amd64
```
Comment (chenzhiwei) [inline deploy/crds/operator.ibm.com_v1alpha1_mongodb_cr.yaml]: can you push the images of three archs to quay.io/opencloudio and build multiarch of them?

make the pullSecret optional or remove it.

**Comment by gyliu513** [comment]
I was asking @morvencao to enable PROW for this repo as well

File: common/manifest.yaml (hunk 1 above)
Comment (gyliu513) [reply common/manifest.yaml]: @chenzhiwei why does not need this?

Comment (chenzhiwei) [reply common/manifest.yaml]: this file is used to build multiarch image, but we use a command that does same:

> manifest-tool push from-args --platforms linux/amd64,linux/ppc64le,linux/s390x --template $(REGISTRY)/$(IMG)-ARCH:$(VERSION) --target $(REGISTRY)/$(IMG) --ignore-missing

**Comment by chenzhiwei** [comment]
@gyliu513 let's merge this PR and then submit follow up PRs to optimize this operator.

/lgtm

**Comment by morvencao** [comment]
@horis233 Can you also add `owner` file in this pr, so that followup pr can be approved?

**Comment by horis233** [comment]
@morvencao Sure, will add

**Pull request merged**